### PR TITLE
[MSPAINT] Make sizeboxes window-less

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Reduce window controls and integrate to the canvas window.
JIRA issue: [CORE-18867](https://jira.reactos.org/browse/CORE-18867)

## Comparison

Win2k3:
image

BEFORE (window hierarchy):
before

AFTER (window hierarchy):
after

## Proposed changes

- The sizeboxes are absorbed by `canvasWindow`.
- class `CSizeboxWindow` is deleted.
- Add `enum CANVAS_HITTEST`.
- Add `getSizeBoxRect`, `getSizeBoxHitTest`, and `drawSizeBoxes` helper functions.

## TODO

- [x] Drawing.
- [x] Zooming.
- [x] Selection.
- [x] Scrolling.
- [x] Move Selection.
- [x] Resize Image.
- [x] Set Cursor.
- [ ] Flicker-less.
- [ ] Resize Selection.
- [ ] Transparent.

### CODE REVIEW

File: base/applications/mspaint/canvas.cpp [hunk 1]
```diff
@@ -142,31 +158,204 @@ LRESULT CCanvasWindow::OnVScroll(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL&
...
             }
+        }
+
+        toolsModel.resetTool();  // resets the point-buffer of the polygon and bezier functions
+        return 0;
+    }
+
+    if (sht == SIZEBOX_CONTENTS)
+    {
+        // TODO:
```
Comment (JoachimHenze) [inline base/applications/mspaint/canvas.cpp:188]: What symptom/bug is to be expected until that TODO will be addressed?

File: base/applications/mspaint/precomp.h [hunk 2]
```diff
@@ -53,6 +53,7 @@
 #include "selection.h"
 #include "selectionmodel.h"
 #include "sizebox.h"
+#include "canvas.h"
```
Comment (JoachimHenze) [inline base/applications/mspaint/precomp.h:56]: What was the motivation to move that include downwards? (Before that it seems likethey were sorted lexicogrpahically).
Don't worry, I wouldn't really mind the moving, I am just curious.

File: base/applications/mspaint/mouse.cpp [hunk 3]
```diff
@@ -17,11 +17,15 @@ POINT ToolBase::pointStack[256] = { { 0 } };
 /* FUNCTIONS ********************************************************/
 
 void
-placeSelWin()
+placeSelWin(void)
```
Comment (JoachimHenze) [inline base/applications/mspaint/mouse.cpp:20]: Superfluous change for cpp code, isn't it?

File: base/applications/mspaint/precomp.h (hunk 2 above)
Comment (katahiromz) [reply base/applications/mspaint/precomp.h:56]: Canvas needs sizeboxes.

File: base/applications/mspaint/canvas.cpp (hunk 1 above)
Comment (katahiromz) [reply base/applications/mspaint/canvas.cpp]:
```txt
// TODO: In future, we handle the events of the window-less image area.
```

File: base/applications/mspaint/mouse.cpp (hunk 3 above)
Comment (katahiromz) [reply base/applications/mspaint/mouse.cpp]: Reverted.

**Pull request merged**